WordPress/Classes/Stores/UserPersistentRepositoryUtility.swift

```diff
@@ -166,12 +167,22 @@ extension UserPersistentRepositoryUtility {
         }
     }
 
-    var isJPContentImportComplete: Bool {
+    var jetpackContentMigrationState: MigrationState {
         get {
-            return UserPersistentStoreFactory.instance().bool(forKey: UPRUConstants.isJPContentImportCompleteKey)
-        }
-        set {
-            UserPersistentStoreFactory.instance().set(newValue, forKey: UPRUConstants.isJPContentImportCompleteKey)
+            let repository = UserPersistentStoreFactory.instance()
+            if let value = repository.string(forKey: UPRUConstants.jetpackContentMigrationStateKey) {
+                return MigrationState(rawValue: value) ?? .notStarted
+            } else if repository.bool(forKey: UPRUConstants.isJPContentImportCompleteKey) {
+                // Migrate the value of the old `isJPContentImportCompleteKey` to `jetpackContentMigrationStateKey`
+                let state = MigrationState.completed
+                repository.set(state.rawValue, forKey: UPRUConstants.jetpackContentMigrationStateKey)
+                repository.set(nil, forKey: UPRUConstants.isJPContentImportCompleteKey)
+                return state
+            } else {
+                return .notStarted
+            }
+        } set {
+            UserPersistentStoreFactory.instance().set(newValue.rawValue, forKey: UPRUConstants.jetpackContentMigrationStateKey)
```

Comment:
This code replaces the old `isJPContentImportComplete` boolean with `jetpackContentMigrationState` of type `MigrationState`. 

To make sure the Migration Flow doesn't appear again for users who already went through the Migration in WPiOS / JPiOS 21.4, this code migrates the existing value of `isJPContentImportComplete` to `jetpackContentMigrationState` ( `true` -> `MigrationState.completed` ).